database/src/pool.rs

```diff
@@ -210,6 +210,26 @@ pub trait Connection: Send + Sync {
         sha: &str,
         parent_sha: &str,
     ) -> anyhow::Result<()>;
+
+    /// Try and mark the benchmark_request as completed. Will return `true` if
+    /// it has been marked as completed else `false` meaning there was no change
+    async fn try_mark_benchmark_request_as_completed(
+        &self,
+        benchmark_request: &mut BenchmarkRequest,
+    ) -> anyhow::Result<bool>;
+
+    /// Given a benchmark request get the id
+    async fn get_benchmark_request_id(
```

Comment:
Tags are unique, let's just use them as the identifier, to avoid the usual ORM issue of "I have a domain struct represented in the DB, how do I attach the ID to it, when it can be None before the struct is inserted into the DB" etc. etc. So we won't need this function, and we can simply use tags when inserting benchmark jobs and marking them as completed.

Reply:
Yup that makes sense to me :+1:. Though still useful at the database level to map to the `benchmark_request` but is more of an internal implementation detail